app/core/utils.js

```diff
@@ -2044,32 +2044,40 @@ const isMobile = () => {
   return mobileRELong.test(ua) || mobileREShort.test(ua.substr(0, 4))
 }
 
-module.exports.getCodeLanguages = () => ({
-  python: {
-    id: 'python',
-    name: `Python (${$.i18n.t('choose_hero.default')})`
-  },
-  javascript: {
-    id: 'javascript',
-    name: 'JavaScript'
-  },
-  coffeescript: {
-    id: 'coffeescript',
-    name: 'CoffeeScript'
-  },
-  lua: {
-    id: 'lua',
-    name: 'Lua'
-  },
-  cpp: {
-    id: 'cpp',
-    name: 'C++'
-  },
-  java: {
-    id: 'java',
-    name: `Java (${$.i18n.t('choose_hero.experimental')})`
+module.exports.getCodeLanguages = () => {
+  const codeLanguages = {
+    python: {
+      id: 'python',
+      name: `Python (${$.i18n.t('choose_hero.default')})`
+    },
+    javascript: {
+      id: 'javascript',
+      name: 'JavaScript'
+    }
   }
-})
+  if (isCodeCombat) {
+    return Object.assign({}, codeLanguages, {
```

Comment:
Not blocking, but why is the first the empty object as the first argument? `return { ...codeLanguages, coffeescript:..., lua:... }` would be cleaner.

Reply:
oh yes. first empty object to prevent `Object.assign` change codeLanguage object